File: src/main/java/com/example/probation/repository/TokenRepository.java
package com.example.probation.repository;

import com.example.probation.core.entity.User;
import com.example.probation.core.entity.VerificationToken;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TokenRepository extends JpaRepository<VerificationToken, Long> {
    Optional<VerificationToken> findByToken(String token);

    Optional<VerificationToken> getByUser(User user);
}

File: src/main/kotlin/com/example/probation/repository/TokenRepository.kt
package com.example.probation.repository

import com.example.probation.core.entity.User
import com.example.probation.core.entity.VerificationToken
import org.springframework.data.jpa.repository.JpaRepository

interface TokenRepository : JpaRepository<VerificationToken, Long> {
    fun findByToken(token: String): VerificationToken?

    fun getByUser(user: User): VerificationToken?

    fun deleteVerificationTokenByUser(user: User)
}
